src/server/scripts/Northrend/AzjolNerub/AzjolNerub/boss_krikthir_the_gatewatcher.cpp

```diff
@@ -88,386 +117,374 @@ class boss_krik_thir : public CreatureScript
-                }
-            }
+    struct npc_watcher_gashraAI : public ScriptedAI
+    {
+        npc_watcher_gashraAI(Creature* creature) : ScriptedAI(creature)
+        {
+            _instance = creature->GetInstanceScript();
+        }
 
-            void UpdateAI(uint32 diff) override
-            {
-                if (!UpdateVictim())
-                    return;
+        void Reset() override
+        {
+            _events.Reset();
+        }
 
-                events.Update(diff);
+        void EnterCombat(Unit* /*who*/) override
+        {
+            _events.ScheduleEvent(EVENT_ENRAGE, randtime(Seconds(3), Seconds(5)));
+            _events.ScheduleEvent(EVENT_WEB_WRAP, randtime(Seconds(16), Seconds(19)));
+            _events.ScheduleEvent(EVENT_INFECTED_BITE, randtime(Seconds(7),Seconds(11)));
+        }
 
-                if (me->HasUnitState(UNIT_STATE_CASTING))
-                    return;
+        void JustDied(Unit* /*killer*/) override
+        {
+            Creature* krikthir = _instance->GetCreature(DATA_KRIKTHIR_THE_GATEWATCHER);
+            if (krikthir && krikthir->IsAlive())
+                krikthir->AI()->DoAction(ACTION_GASHRA_DIED);
+        }
 
-                while (uint32 eventId = events.ExecuteEvent())
-                {
-                    switch (eventId)
-                    {
-                        case EVENT_SUMMON:
-                            Summon();
-                            events.ScheduleEvent(EVENT_SUMMON, 15000);
-                            break;
-                        case EVENT_MIND_FLAY:
-                            DoCastVictim(SPELL_MIND_FLAY);
-                            events.ScheduleEvent(EVENT_MIND_FLAY, 15000);
-                            break;
-                        case EVENT_CURSE_FATIGUE:
-                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 100, true))
-                                DoCast(target, SPELL_CURSE_OF_FATIGUE);
-                            events.ScheduleEvent(EVENT_CURSE_FATIGUE, 10000);
-                            break;
-                        default:
-                            break;
-                    }
-                }
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
 
-                if (!me->HasAura(SPELL_FRENZY) && HealthBelowPct(10))
-                    DoCast(me, SPELL_FRENZY, true);
+            _events.Update(diff);
 
-                DoMeleeAttackIfReady();
-            }
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
 
-            void JustDied(Unit* /*killer*/) override
+            while (uint32 eventId = _events.ExecuteEvent())
             {
-                Talk(SAY_DEATH);
-                _JustDied();
+                switch (eventId)
+                {
+                    case EVENT_ENRAGE:
+                        DoCast(nullptr, SPELL_ENRAGE);
```

Comment:
`DoCast` with a single argument is a [pretty weird overload](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/AI/CoreAI/UnitAI.cpp#L135).